app/web/pages/leave-reference/[[...slug]].tsx

```diff
@@ -0,0 +1,48 @@
+import { appGetLayout } from "components/AppRoute";
+import NotFoundPage from "features/NotFoundPage";
+import LeaveReferencePageComponent from "features/profile/view/leaveReference/LeaveReferencePage";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { useRouter } from "next/router";
+import nextI18nextConfig from "next-i18next.config";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { referenceTypeRouteStrings } from "routes";
+
+export const getStaticPaths: GetStaticPaths = () => ({
+  paths: [],
+  fallback: "blocking",
+});
+
+export const getStaticProps: GetStaticProps = async ({ locale }) => ({
+  props: {
+    ...(await serverSideTranslations(
+      locale ?? "en",
+      ["global", "profile"],
+      nextI18nextConfig
+    )),
+  },
+});
+export default function LeaveReferencePage() {
+  const router = useRouter();
+
+  // leave-reference/:type/:userId/:hostRequestId?
+  const slug = router.query.slug;
+  if (!slug?.[0] || slug?.[1]) return <NotFoundPage />;
```

Comment:
```suggestion
  if (!slug?.[0] || !slug?.[1]) return <NotFoundPage />;
```
Otherwise this was sending valid `userId` to the not found page

Reply:
But beyond that, this whole reference leaving thing is broken after I changed the above :/ And most of it is because of the routing mess... we're probably not making the logic here any easier by having `step` as `slug[3]` if it's a friend reference but `slug[4]` if it's a host/surfer reference.

I think the URL should reference the request ID if this is available tbh so it can be unified as `/leave-reference/[reference-id]/[step]`. Maybe a future thing to change unless fixing that is not enough? Haven't delved deeper beyond that.